server/service/src/programs/update_program_document.rs

```diff
@@ -173,19 +170,19 @@ fn is_truthy(value: &Value) -> bool {
     return true;
 }
 
-fn match_condition(condition: &EventCondition, doc: &RawDocument) -> bool {
+fn match_condition(condition: &EventCondition, doc: &Document) -> bool {
     let Some(field) = extract_value_field(&doc.data, &condition.field) else {
-        if condition.is_falsy.is_some() {
+        if condition.is_falsy.unwrap_or(false) {
```

Comment:
hmm, not really sure how this change relates to sync, but looks like it's a fix for something ? Just wondering could we catch this with tests ?